tests/mozdef_util/test_elasticsearch_client.py

```diff
@@ -120,6 +120,27 @@ def test_search_no_results(self):
         assert results['hits'] == []
 
 
+class TestCloseIndex(ElasticsearchClientTest):
+
+    def test_index_close(self):
+        if pytest.config.option.delete_indexes:
+            self.es_client.create_index('test_index')
+        time.sleep(1)
+        indices = self.es_client.index_close('test_index')
```

Comment:
I think these tests can be improved to verify more of the open/close logic. All we are doing here is checking that we can run 2 functions, and then that the second function, returns True.

For example, what if we closed the events index, and then tried to save an event. When you try writing to a closed index, elasticsearch throws an exception, so we can tell pytest to expect an exception like https://github.com/mozilla/MozDef/blob/master/tests/mozdef_util/query_models/test_search_query.py#L291

Reply:
The first test closes the test index tha L#127 creates.
The next test creates a test index on L#137 and then closes it on L#139, and then opens the one we closed on L#140.

Are you just asking for a negative test case for saving to a closed index? That will never happen in production as we only save to the current events alias, which is never closed. We only close  "actual index names" not aliases. In fact in 6.x you can't close an alias at all, you MUST cite the index name.  We should never, ever, be writing to any indexes that are not tagged with the current events alias.

Please help me find the value in this request or explain how it will benefit us moving forward.

Reply:
What I'm saying is, you aren't actually testing the functionality that an index does get closed, you're testing that `self.es_client.index_close('test_index')` returns a True value.

For example, if this is what the implementation looks like of index_close, your tests will pass, which is not what we want:
```
def index_close(index_name):
    return True
```

So, to improve these tests, we want to have the tests verify that `self.es_client.index_close('test_index')` actually closes an index (that that function is doing what we want it to do). 

One way to verify if an index actually gets closed, is to try and write to it and expecting an elasticsearch closed index write error to get thrown.

Another idea might be to query elasticsearch and have them tell us if the `test_index` is now closed.

Reply:
I understand now, thank you for clarifying.